### PR TITLE
microarchitectures: only call sysctl once on macOS

### PR DESCRIPTION
**Pull request by tgamblin**
The original code called the `sysctl` command 10 times to get 5 pieces of information, but you can get the same stuff with only one external process launch.  Just call it once to reduce Spack startup time.

- [x] coalesce 5 calls to `sysctl` into one
- [x] memoize call to `raw_info_dict`

### CODE REVIEW

**Comment by tgamblin** [comment]
`sysctl_info_dict` is still called twice with the same args here... digging around to figure out why.

**Comment by tgamblin** [comment]
ok pretty sure this one is done.  `raw_info_dict()` is now memoized.

File: lib/spack/llnl/util/cpu/detect.py [hunk 1]
```diff
@@ -101,13 +101,22 @@ def sysctl(*args):
             ['sysctl'] + list(args), env=child_environment
         ).strip()
 
-    flags = (sysctl('-n', 'machdep.cpu.features').lower() + ' '
-             + sysctl('-n', 'machdep.cpu.leaf7_features').lower())
+    # get all the sysctl info we need in one call
+    print("CALLING")
```
Comment (adamjstewart) [inline lib/spack/llnl/util/cpu/detect.py]: Need to remove this print statement

**Comment by tgamblin** [comment]
I'll have to figure out why this slows things down.  I'd be surprised if memoization were so slow, but maybe it is.

**Pull request closed without merging**